src/main/java/course_project/Main.java

```diff
@@ -0,0 +1,10 @@
+package course_project;
+
+import course_project.util.Launcher;
+
+public class Main {
+
+    public static void main(String[] args) {
+        Launcher.launch();
```

Comment:
In case of destroyed player's ship, cells around are not marked as miss. Bug or feature?
![image](https://user-images.githubusercontent.com/58872509/134482935-36166385-e5c1-484c-a8fe-ff18148f7f3b.png)

Reply:
Bug in method markDestroyedShip() 😞

Reply:
Good job! The game interface is nice, it's comfortable to play! Automatic ship placement is cool, the game has nice features, messages are pretty informative. There's some architecture, but model/service separation is not met. There are nice abstractions, but some methods seem to be oversized. Code in general is easy to read. Also, you show good knowledge of Java Core and Java libraries, missing some benefits of OOP - like introducing interfaces/abstract classes. DRY is not always followed. 

In general, the app leaves a good impression. Approved!

Reply:
Thanks!